lib/literal/array.rb

```diff
@@ -77,6 +79,39 @@ def &(other)
 		end
 	end
 
+	def *(times)
+		case times
+		when String
+			@__value__ * times
+		else
+			__with__(@__value__ * times)
+		end
+	end
+
+	def +(other)
+		case other
+		when ::Array
+			values = @__value__ + @__generic__.new(*other).__value__
```

Comment:
This should probably check other against the collection type and then use `__with__` to return a Literal::Array. I believe this will return an Array at the moment.

Reply:
This creates a temporary variable `values` (perhaps poorly named, too close to `__values__`) that is used to initialize a new `Literal::Array` on line 101. So the type checking is deferred to the instantiation of the new `Literal::Array` at the end of the method.

I think my error here is that I'm creating a new Literal::Array twice. This should work as just:

```
values = @__value__ + other
```

And the return value could actually be:

```
@__generic__.new(*values)
```

Does this approach make sense; should we be checking types inside the `case...when`?